modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/render.es.js

```diff
@@ -71,7 +73,9 @@ export default function render(renderable, renderData, container) {
 		// eslint-disable-next-line @liferay/portal/no-react-dom-render
 		ReactDOM.render(
 			<ClayIconSpriteContext.Provider value={spritemap}>
-				{Component ? <Component {...renderData} /> : renderable}
+				<A11y enable={process.env.NODE_ENV === 'development'}>
+					{Component ? <Component {...renderData} /> : renderable}
+				</A11y>
```

Comment:
I agree with your point about the performance ramifications of this. This whole thing seems like something that would be useful not just to us internally, but also to customers, which suggests that it should be something that has a GUI to enable it on some basis (ie. a site admin should be able to browse even the production site with this turned on in order to detect issues; but everybody else should pay zero cost, not see the feature, and not not download any of related code). That's obviously a "product owner" type call so it would be good to get some input from @dsanz and @jbalsas on that.

Reply:
This LPS is just to add the ability to check a page for a11y issues, it's unrelated to DXP itself, except that it must hook at some point in the render process. It's planned to have more features on top of this. For instance, like Greg mentioned, some tool that would be enabled for all pages by default in a particular build created for that purpose, but that tool would not be shipped to customers. Similarly, it's also planned to apply the tool only for content inputted by DXP users, with some integration with CKEditor

Reply:
Hmm now I see that we have several plans for this component 😁!